core/src/main/java/org/kiwix/kiwixmobile/core/dao/NewRecentSearchDao.kt

```diff
@@ -40,6 +40,7 @@ class NewRecentSearchDao @Inject constructor(
   }
 
   fun saveSearch(title: String, id: String) {
+    deleteSearchString(title)
```

Comment:
I would have just changed `distinct` to `distinctBy{::searchTerm}` in `recentSearches(zimId: String?)`.

This is technically incorrect because we retrieve search results by the id of a book as well as a term but this would delete search terms across all books, were there duplicates. This is the fault of the delete function to be fair.